### PR TITLE
FIX: routing on /new-topic and /categories

### PR DESCRIPTION
**Pull request by renato**
This is very hard to test, it was happening very intermittently:
* When visiting `/new-topic` with category and tags params, _sometimes_ it would render the wrong app, the 1st option instead of the one inferred from the tags param
* When visiting `/` or `/categories` after having visited a different global filter (which is stored as a preference), the presented data would be from the preferred app, but the "visually" selected app was the 1st option in the list

### CODE REVIEW

**Review by pmusaraj — APPROVED** [review]
Looks good to me, approving. Will ping Isaac as well given his experience with this plugin.

File: assets/javascripts/discourse/initializers/apply-global-filter-styles.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ export default {
 
     if (!tags) {
       // Use first global filter as default
```
Comment (janzenisaac) [inline assets/javascripts/discourse/initializers/apply-global-filter-styles.js:46]:
```suggestion
```

We can drop this comment, I don't think it is true anymore.

File: assets/javascripts/discourse/initializers/global-filter-preference.js [hunk 2]
```diff
@@ -57,7 +57,6 @@ export default {
 
           if (tagFromNewTopic) {
             this._setClientFilterPref(tagFromNewTopic, currentUser);
```
Comment (janzenisaac) [inline assets/javascripts/discourse/initializers/global-filter-preference.js:59]: Not returning from this condition means that we may pass the condition of `ROUTES_TO_REDIRECT_ON` and have an additional redirect... this makes me nervous. Have you experienced any adverse effects / redirects when testing this on any of the `ROUTES_TO_REDIRECT_ON`?

Comment (renato) [reply assets/javascripts/discourse/initializers/global-filter-preference.js:59]: I haven't experienced any adverse effects so far.

the issue is that around 1 out of 8 times I get the wrong app with the `return`, running on `localhost`. On a production instance it's always going to the wrong app

**Review by janzenisaac — APPROVED** [review]

**Pull request merged**